src/index.ts

```diff
@@ -525,9 +523,8 @@ export class FSWatcher extends EventEmitter {
 
   /**
    * Close watchers and remove all listeners from watched paths.
-   * @returns {Promise<void>}.
    */
-  close() {
+  close(): Promise<void> | undefined {
```

Comment:
@benmccann `this._closePromise` may be undefined, should we return undefined as well, or use a non-null assertion?

Reply:
I didn't get to look at this before it was merged, but to answer your question I think `Promise<void>` is the correct return type here and we should not return `undefined`. I've opened a new issue to discuss it: https://github.com/paulmillr/chokidar/issues/1357